platform-api/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/stack/StackLoader.java

```diff
@@ -0,0 +1,138 @@
+                                                      .registerTypeAdapter(ProjectConfig.class, new ProjectConfigSerializer())
+                                                      .registerTypeAdapter(EnvironmentState.class, new EnvironmentStateSerializer())
+                                                      .registerTypeAdapter(Command.class, new CommandSerializer())
+                                                      .registerTypeAdapter(Recipe.class, new RecipeSerializer())
+                                                      .registerTypeAdapter(Limits.class, new LimitsSerializer())
+                                                      .registerTypeAdapter(MachineSource.class, new MachineSourceSerializer())
+                                                      .registerTypeAdapter(MachineConfig.class, new MachineSourceSerializer())
+                                                      .registerTypeAdapter(StackSource.class, new StackSourceSerializer())
+                                                      .registerTypeHierarchyAdapter(byte[].class, new ByteArrayToBase64Serializer())
+                                                      .registerTypeAdapter(Permissions.class, new PermissionsSerializer())
+                                                      .registerTypeAdapter(Group.class, new GroupSerializer())
+                                                      .create();
+
+    private final Set<String> stackPaths;
+    private final StackDao    stackDao;
+
+    @Inject
+    public StackLoader(@Named("predefined.stack.path") Set<String> stackPaths, StackDao stackDao) {
+        this.stackPaths = firstNonNull(stackPaths, Collections.<String>emptySet());
+        this.stackDao = stackDao;
+    }
+
+    @PostConstruct
+    public void start() throws ServerException {
+        for (String stackPath: stackPaths) {
+            if (!Strings.isNullOrEmpty(stackPath)) {
+                for (StackImpl stack: loadStacks(stackPath)) {
+                    try {
+                        try {
+                            stackDao.update(stack);
+                        } catch (NotFoundException e) {
+                            stackDao.create(stack);
+                        }
+                    } catch (ConflictException e) {
+                        throw new ServerException("Failed to store stack " + stack, e);
+                    }
+                }
+            }
+        }
+    }
+
+    private List<StackImpl> loadStacks(String stackPath) throws ServerException {
+        try (InputStream is = getResource(stackPath)) {
+            return firstNonNull(GSON.fromJson(new InputStreamReader(is), new TypeToken<List<StackImpl>>() {}.getType()), emptyList());
+        } catch (IOException | JsonIOException | JsonSyntaxException e) {
+            throw new ServerException("Failed to get stacks from specified path " + stackPath, e);
+        }
+    }
+
+    private InputStream getResource(String resource) throws IOException {
```

Comment:
Consider to use this class http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/io/Resources.html 
instead of this method